### PR TITLE
chore: experimental w4p support

### PR DESCRIPTION
**Pull request by nferch**

### CODE REVIEW

File: packages/sdk/client/src/services/userfunctions.ts
[the remark this answers is not in the document]
Comment (nferch) [reply packages/sdk/client/src/services/userfunctions.ts:1]: Seemed like the least bad place to put it for now, it's alongside the Agent Hosting client which Rich had suggested live there.

[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/sdk/client/src/services/userfunctions.ts:1]: functions is a pretty app-centric thing, I don't think it's something that the client should know about. I'd suggest all code related to this live in plugins for now.

[the remark this answers is not in the document]
Comment (nferch) [reply packages/sdk/client/src/services/userfunctions.ts:1]: how would a CLI use code in a plugin?

[the remark this answers is not in the document]
Comment (wittjosiah) [reply packages/sdk/client/src/services/userfunctions.ts:1]: maybe we end up doing this in a package like `@dxos/functions` or something, but functions are a higher level thing than client and I don't think we should leak functions into client.

absent another package we'd likely end up with just another entrypoint to the plugin (this is what we're planning to do with the schemas)

[the remark this answers is not in the document]
Comment (richburdon) [reply packages/sdk/client/src/services/userfunctions.ts:1]: is this functions specific or edge-related -- i.e., auth/protobuf stuff for all edge services. if so, i'd suggest a new @dxos/edge package.

**Pull request merged**